clevertap/src/main/java/com/rudderstack/android/integrations/clevertap/CleverTapIntegrationFactory.java

```diff
@@ -1,47 +1,398 @@
+    );
+
+    public static Factory FACTORY = new Factory() {
+        @Override
+        public RudderIntegration<?> create(
+                Object settings,
+                RudderClient client,
+                RudderConfig rudderConfig
+        ) {
+            return new CleverTapIntegrationFactory(settings, rudderConfig);
+        }
+
+        @Override
+        public String key() {
+            return CLEVERTAP_KEY;
+        }
+    };
+
+    private CleverTapIntegrationFactory(
+            Object config,
+            RudderConfig rudderConfig
     ) {
-      return new CleverTapIntegrationFactory(settings, rudderConfig);
+        if (RudderClient.getApplication() == null) {
+            RudderLogger.logError(
+                    "Application is null. Aborting CleverTap initialization."
+            );
+            return;
+        }
+
+        Gson gson = new Gson();
+        CleverTapDestinationConfig destinationConfig = gson.fromJson(
+                gson.toJson(config),
+                CleverTapDestinationConfig.class
+        );
+        if (TextUtils.isEmpty(destinationConfig.accountToken) || TextUtils.isEmpty(destinationConfig.accountId)) {
+            RudderLogger.logError("Invalid CleverTap Account Credentials, Aborting");
+            return;
+        }
+        if (!destinationConfig.region.equals("none")) {
+            CleverTapAPI.changeCredentials(
+                    destinationConfig.accountId,
+                    destinationConfig.accountToken,
+                    destinationConfig.region
+            );
+        } else {
+            CleverTapAPI.changeCredentials(
+                    destinationConfig.accountId,
+                    destinationConfig.accountToken
+            );
+        }
+        cleverTap = CleverTapAPI.getDefaultInstance(RudderClient.getApplication());
+        if (rudderConfig.getLogLevel() >= RudderLogger.RudderLogLevel.DEBUG) {
+            CleverTapAPI.setDebugLevel(CleverTapAPI.LogLevel.DEBUG);
+        } else {
+            CleverTapAPI.setDebugLevel(CleverTapAPI.LogLevel.INFO);
+        }
+        RudderLogger.logInfo("Initialized Clever Tap SDK");
+
+        RudderClient
+                .getApplication()
+                .registerActivityLifecycleCallbacks(
+                        new Application.ActivityLifecycleCallbacks() {
+                            @Override
+                            public void onActivityCreated(
+                                    Activity activity,
+                                    Bundle savedInstanceState
+                            ) {
+                                if (cleverTap == null) return;
+                                CleverTapAPI.setAppForeground(true);
+                                try {
+                                    cleverTap.pushNotificationClickedEvent(activity.getIntent().getExtras());
+                                } catch (Exception e) {
+                                }
```

Comment:
should this be empty?

Reply:
Yeah its just that we will not break if the above action crashes

Reply:
ok..